pkg/grub/rootfs.cfg

```diff
@@ -452,6 +452,32 @@ function set_isolcpus {
    set_global dom0_extra_args "$dom0_extra_args isolcpus=inverse,0 nohz_full=inverse,0"
 }
 
+# Read eve-kernel-extra-cmdline from EFI vars and append to dom0_extra_args
+function set_append_extra_efi_cmdline {
+    insmod efi
+    insmod getenv
+
+    # Must match the GUID used by pillar
+    set _eve_guid=7AD58F29-2B49-4F5A-9F0B-4E7BF7C2C311
+
+    # Put efivar payload into a temp var (ASCII+NUL expected)
+    unset eve_kcmdline
+    if getenv -e eve-kernel-extra-cmdline -g $_eve_guid eve_kcmdline; then
+        if [ -n "$eve_kcmdline" ]; then
+            # Append with a single space if dom0_extra_args already set
+            if [ -n "$dom0_extra_args" ]; then
+                set_global dom0_extra_args "$dom0_extra_args $eve_kcmdline"
```

Comment:
Can we use it in both cases?...

Reply:
@OhmSpectator it can add extra leading space